app/main.py

```diff
@@ -1 +1,46 @@
-#  write your code here
+
+class Animal:
+    def __init__(self, name: str,
+                 appetite: int,
+                 is_hungry : bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            self.is_hungry = False
+            print(f"Eating {self.appetite} food points...")
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+    def __init__(self, name : str,
+                 is_hungry : bool = True) -> None:
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name : str,
+                 is_hungry : bool = True) -> None:
+        super().__init__(name, appetite=7, is_hungry=True)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals : list) -> None:
+    total_food = 0
+    for animal in animals:
+        eaten_food = animal.feed()
+        total_food += eaten_food
+    return total_food
```

Comment:
You can add a type of objects in list.
Also, please read a checklist before pushing your solution.
Point 2 in checklist: Don't use a for loop if you can use generator expression.